dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueChannel.java

```diff
@@ -242,6 +242,8 @@ private static ImmutableList<LimitedChannel> createHostChannels(
                 channel = HostMetricsChannel.create(cf, channel, targetUri.uri());
                 channel =
                         new TraceEnrichingChannel(channel, DialogueTracing.tracingTags(cf, uriIndexForInstrumentation));
+                channel = new DeadlineAdvertisementChannel(
+                        channel, cf.clientConf().readTimeout());
```

Comment:
There are a couple places which set a read timeout value of zero expecting it to be infinite. We handle this case in the apache httpclient shim here: https://github.com/palantir/dialogue/blob/f4915fc9eee84d202ed1102c41af31995594d670/dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java#L641-L648
I think we'll need to handle that case in the DeadlineAdvertisementChannel

Reply:
👍 fixed!